plugins/src/core/index.ts

```diff
@@ -0,0 +1,369 @@
+} from "./handlers";
+
+export interface OptimoveConfig{
+  pushReceivedHandler?: PushNotificationHandler | null;
+  pushOpenedHandler?: PushNotificationHandler | null;
+  inAppDeepLinkHandler?: InAppDeepLinkHandler | null;
+  inAppInboxUpdatedHandler?: InAppInboxUpdatedHandler | null;
+  deepLinkHandler?: DeepLinkHandler | null;
+}
+
+let currentConfig: OptimoveConfig = {
+  pushReceivedHandler: null,
+  pushOpenedHandler: null,
+  inAppDeepLinkHandler: null,
+  inAppInboxUpdatedHandler: null,
+  deepLinkHandler: null,
+};
+document.addEventListener("deviceready", init, false);
+document.addEventListener("resume", resume, false);
+document.addEventListener("pause", pause, false);
+
+function init() {
+  setContext();
+}
+
+function resume() {
+  setContext();
+}
+
+function pause() {
+  clearContext();
+}
+
+function setContext() {
+  setHandlersCallBackContext().then(
+    (successMessage) => {
+      console.log(successMessage);
+    },
+    (errorMessage) => {
+      console.error(errorMessage);
+    }
+  );
+}
+function clearContext() {
+  return new Promise((resolve, reject) => {
+    cordova.exec(resolve, reject, "OptimoveSDKPlugin", "clearContext", []);
+  });
+}
+
+function setHandlersCallBackContext() {
+  return new Promise((resolve, reject) => {
+    cordova.exec(
+      nativeMessageHandler,
+      reject,
+      "OptimoveSDKPlugin",
+      "setHandlersCallBackContext",
+      []
+    );
+  });
+}
+function checkIfPendingDDLExists() {
+  return new Promise((resolve, reject) => {
+    cordova.exec(
+      nativeMessageHandler,
+      reject,
+      "OptimoveSDKPlugin",
+      "checkIfPendingDDLExists",
+      []
+    );
+  });
+}
+
+function checkIfPendingPushExists() {
+  return new Promise((resolve, reject) => {
+    cordova.exec(
+      nativeMessageHandler,
+      reject,
+      "OptimoveSDKPlugin",
+      "checkIfPendingPushExists",
+      []
+    );
+  });
+}
+
+function nativeMessageHandler(message) {
+  if (!message || typeof message === "string") {
+    return;
+  }
+
+  const handlerName = `${message.type}Handler`;
+  if (
+    handlerName === "inAppInboxUpdatedHandler" &&
+    typeof currentConfig[handlerName] === "function"
+  ) {
+    currentConfig[handlerName]?.();
+
+    return;
+  }
+
+  if (typeof currentConfig[handlerName] === "function") {
+    currentConfig[handlerName](message.data);
+  } else {
+    console.log(`Optimove: No handler defined for '${message.type}' event`);
+  }
+}
+
+interface InAppInboxItem {
+  id: number;
+  title: string;
+  subtitle: string;
+  availableFrom: string | "";
+  availableTo: string | "";
+  dismissedAt: string | "";
+  isRead: boolean;
+  sentAt?: string;
+  data?: { [key: string]: any };
+  imageUrl?: string;
+}
+interface EventParams {
+  [key: string]: any;
+}
+interface InAppInboxSummary {
+  totalCount: number;
+  unreadCount: number;
+}
+
+const Optimove = {
+  /**
+   * Sets the User ID of the current user and starts the {@code Visitor} to {@code Customer} conversion flow.
```

Comment:
are these descriptions taken from somewhere? Do consumers of the SDK really need to know about `starts the {@code Visitor} to {@code Customer} conversion flow.`?

Reply:
I used Android SDK and Kumulos Cordova SDK as a reference

Reply:
y, but java annotations dont belong to typescript

Reply:
same for @link